pkg/storage/secret/metadata/keeper_store.go

```diff
@@ -361,7 +368,7 @@ func (s *keeperMetadataStorage) validateSecureValueReferences(ctx context.Contex
 		// We are guaranteed that the returned `secureValueRows` are a subset of `usedSecureValues`,
```

Comment:
Another small issue I found was that if the same `secureValueName` was used multiple times in the same Keeper, we would enter this condition no matter what, because the values in `usedSecureValues` were duplicated, so I changed it to a map to gather unique names.